docs/src/index.md

```diff
@@ -5,3 +5,98 @@
 [![codecov.io](http://codecov.io/github/invenia/Nabla.jl/coverage.svg?branch=master)](http://codecov.io/github/invenia/Nabla.jl?branch=master)
 [![Stable Docs](https://img.shields.io/badge/docs-stable-blue.svg)](https://invenia.github.io/Nabla.jl/stable)
 [![Latest Docs](https://img.shields.io/badge/docs-latest-blue.svg)](https://invenia.github.io/Nabla.jl/latest)
+
+# Getting Started
+
+`Nabla.jl` has two interfaces, both of which we expose to the end user. We first provide a minimal working example with the high-level interface, and subsequently show how the low-level interface can be used to achieve similar results. More involved examples can be found [here](https://github.com/invenia/Nabla.jl/tree/master/examples).
+
+## A Toy Problem
+
+Consider the gradient of a vector-quadratic function. The following code snippet constructs such a function, and inputs `x` and `y`.
+```@example toy
+using Nabla
+
+# Generate some data.
+rng, N = MersenneTwister(123456), 2
+x, y = randn.(rng, [N, N])
+A = randn(rng, N, N)
+
+# Construct a vector-quadratic function in `x` and `y`.
+f(x, y) = y.' * (A * x)
+f(x, y)
+```
+
+Only a small amount of [matrix calculus](https://en.wikipedia.org/wiki/Matrix_calculus) is required to the find the gradient of `f(x, y)` w.r.t. `x` and `y`, which we denote by `∇x` and `∇y` respectively, to be
+
+```@example toy
+(∇x, ∇y) = (A.'y, A * x)
+```
+
+## High-Level Interface
+The high-level interface provides a simple way to "just get the gradients" w.r.t. each argument of `f`:
+```@example toy
+∇x, ∇y = ∇(f)(x, y)
+```
+This interface is implemented in `core.jl`, and is a thin wrapper of the low-level interface constructed above. Here, we first use `∇` to get a function which, when evaluated, returns the gradient of `f` w.r.t. each of it's inputs at the values of the inputs provided.
+
+We may provide an optional argument to also return the value `f(x, y)`:
+```@example toy
+(z, (∇x, ∇y)) = ∇(f, true)(x, y)
+```
+
+If the gradient w.r.t. a single argument is all that is required, or a subset of the arguments for an N-ary function, we recommend closing over the arguments which respect to which you do not wish to take gradients. For example, to take the gradient w.r.t. just `x`, one could do the following:
```

Comment:
Even though the examples illustrate so, it might be good to explicitly remark that taking the gradient of a single-output functions yields a _1-tuple_ containing the gradient.